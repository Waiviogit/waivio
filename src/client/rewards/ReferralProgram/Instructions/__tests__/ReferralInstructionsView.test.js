import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { IntlProvider } from 'react-intl';
import { shallow, mount } from 'enzyme';
import {
  mockDataConfirm,
  mockDataIsBlackListUser,
  mockDataIsLoadingState,
  mockDataView,
} from '../__mock__/mockData';
import ReferralsInstructionsView from '../../Instructions/ReferralInstructionsView';

describe('ReferralInstructionsView', () => {
  const handleCopyTextButton = jest.fn();
  const widget = `<iframe></iframe>`;

  let props;
  let wrapper;

  beforeEach(() => {
    props = mockDataView;
    wrapper = shallow(
      <BrowserRouter>
        <IntlProvider locale="en">
          <ReferralsInstructionsView
            mainProps={props}
            handleCopyTextButton={handleCopyTextButton}
            widget={widget}
          />
        </IntlProvider>
      </BrowserRouter>,
    );
  });

  afterEach(() => jest.clearAllMocks());

  it('should create snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should simulate click on copy-icon', () => {
    props = mockDataConfirm;
    wrapper = mount(
      <BrowserRouter>
        <IntlProvider locale="en">
          <ReferralsInstructionsView
            mainProps={props}
            handleCopyTextButton={handleCopyTextButton}
            widget={widget}
          />
        </IntlProvider>
      </BrowserRouter>,
    );
    const container = wrapper.find('Icon');
    container.simulate('click');
    expect(handleCopyTextButton).toHaveBeenCalled();
  });

  it('should return info for blacklist user', () => {
    props = mockDataIsBlackListUser;
    wrapper = mount(
      <BrowserRouter>
        <IntlProvider locale="en">
          <ReferralsInstructionsView
            mainProps={props}
            handleCopyTextButton={handleCopyTextButton}
            widget={widget}
          />
        </IntlProvider>
      </BrowserRouter>,
    );
    const container = wrapper.find('.ReferralInstructions__is-blacklist').props().children.props
      .defaultMessage;
    expect(container).toBe(
      'Your account {username} is listed in the Waivioâ€™s blacklist or in other blacklists trusted by Waivio and you are not eligible to participate in the Referral program.',
    );
  });

  it('should return loading state', () => {
    props = mockDataIsLoadingState;
    wrapper = mount(
      <BrowserRouter>
        <IntlProvider locale="en">
          <ReferralsInstructionsView
            mainProps={props}
            handleCopyTextButton={handleCopyTextButton}
            widget={widget}
          />
        </IntlProvider>
      </BrowserRouter>,
    );
    const container = wrapper.find('.ReferralInstructions__wrap-conditions__loader').props()
      .children.props;
    expect(container).toEqual({ center: true });
  });
});
