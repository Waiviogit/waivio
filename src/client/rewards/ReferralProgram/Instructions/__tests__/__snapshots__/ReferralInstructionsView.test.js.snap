// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReferralInstructionsView should create snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BrowserRouter>
    <IntlProvider
      locale="en"
    >
      <ReferralsInstructionsView
        handleCopyTextButton={[MockFunction]}
        mainProps={
          Object {
            "authUserName": "vallon",
            "currentCopyText": "",
            "currentStatus": false,
            "handleAgreeRulesCheckbox": [MockFunction],
            "handleCancelButton": [MockFunction],
            "handleOkButton": [MockFunction],
            "isAuthenticated": true,
            "isBlackListUser": false,
            "isGuest": false,
            "isModal": false,
            "isStartChangeRules": false,
            "isStartGetReferralInfo": false,
            "rejectRules": [MockFunction],
            "setIsCopyButton": [MockFunction],
          }
        }
        widget="<iframe></iframe>"
      />
    </IntlProvider>
  </BrowserRouter>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <IntlProvider
        locale="en"
      >
        <ReferralsInstructionsView
          handleCopyTextButton={[MockFunction]}
          mainProps={
            Object {
              "authUserName": "vallon",
              "currentCopyText": "",
              "currentStatus": false,
              "handleAgreeRulesCheckbox": [MockFunction],
              "handleCancelButton": [MockFunction],
              "handleOkButton": [MockFunction],
              "isAuthenticated": true,
              "isBlackListUser": false,
              "isGuest": false,
              "isModal": false,
              "isStartChangeRules": false,
              "isStartGetReferralInfo": false,
              "rejectRules": [MockFunction],
              "setIsCopyButton": [MockFunction],
            }
          }
          widget="<iframe></iframe>"
        />
      </IntlProvider>,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ReferralsInstructionsView
          handleCopyTextButton={[MockFunction]}
          mainProps={
            Object {
              "authUserName": "vallon",
              "currentCopyText": "",
              "currentStatus": false,
              "handleAgreeRulesCheckbox": [MockFunction],
              "handleCancelButton": [MockFunction],
              "handleOkButton": [MockFunction],
              "isAuthenticated": true,
              "isBlackListUser": false,
              "isGuest": false,
              "isModal": false,
              "isStartChangeRules": false,
              "isStartGetReferralInfo": false,
              "rejectRules": [MockFunction],
              "setIsCopyButton": [MockFunction],
            }
          }
          widget="<iframe></iframe>"
        />,
        "locale": "en",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleCopyTextButton": [MockFunction],
          "mainProps": Object {
            "authUserName": "vallon",
            "currentCopyText": "",
            "currentStatus": false,
            "handleAgreeRulesCheckbox": [MockFunction],
            "handleCancelButton": [MockFunction],
            "handleOkButton": [MockFunction],
            "isAuthenticated": true,
            "isBlackListUser": false,
            "isGuest": false,
            "isModal": false,
            "isStartChangeRules": false,
            "isStartGetReferralInfo": false,
            "rejectRules": [MockFunction],
            "setIsCopyButton": [MockFunction],
          },
          "widget": "<iframe></iframe>",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <IntlProvider
          locale="en"
        >
          <ReferralsInstructionsView
            handleCopyTextButton={[MockFunction]}
            mainProps={
              Object {
                "authUserName": "vallon",
                "currentCopyText": "",
                "currentStatus": false,
                "handleAgreeRulesCheckbox": [MockFunction],
                "handleCancelButton": [MockFunction],
                "handleOkButton": [MockFunction],
                "isAuthenticated": true,
                "isBlackListUser": false,
                "isGuest": false,
                "isModal": false,
                "isStartChangeRules": false,
                "isStartGetReferralInfo": false,
                "rejectRules": [MockFunction],
                "setIsCopyButton": [MockFunction],
              }
            }
            widget="<iframe></iframe>"
          />
        </IntlProvider>,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <ReferralsInstructionsView
            handleCopyTextButton={[MockFunction]}
            mainProps={
              Object {
                "authUserName": "vallon",
                "currentCopyText": "",
                "currentStatus": false,
                "handleAgreeRulesCheckbox": [MockFunction],
                "handleCancelButton": [MockFunction],
                "handleOkButton": [MockFunction],
                "isAuthenticated": true,
                "isBlackListUser": false,
                "isGuest": false,
                "isModal": false,
                "isStartChangeRules": false,
                "isStartGetReferralInfo": false,
                "rejectRules": [MockFunction],
                "setIsCopyButton": [MockFunction],
              }
            }
            widget="<iframe></iframe>"
          />,
          "locale": "en",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleCopyTextButton": [MockFunction],
            "mainProps": Object {
              "authUserName": "vallon",
              "currentCopyText": "",
              "currentStatus": false,
              "handleAgreeRulesCheckbox": [MockFunction],
              "handleCancelButton": [MockFunction],
              "handleOkButton": [MockFunction],
              "isAuthenticated": true,
              "isBlackListUser": false,
              "isGuest": false,
              "isModal": false,
              "isStartChangeRules": false,
              "isStartGetReferralInfo": false,
              "rejectRules": [MockFunction],
              "setIsCopyButton": [MockFunction],
            },
            "widget": "<iframe></iframe>",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
